src/app.js

```diff
@@ -1,108 +1,71 @@
-export function app(app) {
-  var state = {}
+export function app(props) {
+  var state
   var actions = {}
   var events = {}
   var mixins = []
-  var view = app.view
-  var root = app.root || document.body
+  var view = props.view
   var node
-  var element
-  var locked = false
-  var loaded = false
+  var root = props.root
+  var host = (root && root.parentNode) || document.body
+  var lock = false
```

Comment:
could we just do `var lock` with no `false`?  `undefined` will coerce to `true`, and it will just cycle between the booleans from there.